### PR TITLE
Refactor conversation model

### LINKED ISSUE #60: Fix "Rubocop/Metrics/MethodLength" issue in app/models/conversation.rb
Method has too many lines. [15/10]

https://codeclimate.com/github/chatwoot/chatwoot/app/models/conversation.rb#issue_5d87480c87cf1900010003c5

### LINKED ISSUE #61: Fix "Rubocop/Metrics/MethodLength" issue in app/models/conversation.rb
Method has too many lines. [14/10]

https://codeclimate.com/github/chatwoot/chatwoot/app/models/conversation.rb#issue_5d87480c87cf1900010003c7

### PR DESCRIPTION
**Pull request by MatheusRich**
This will fix several issues.

Fixes #60
Fixes #61

### CODE REVIEW

**Comment by MatheusRich** [comment]
Merging #106 before would be good too

**Comment by pranavrajs** [comment]
@MatheusRich I've merged #106 Could you please rebase the branch?

**Comment by MatheusRich** [comment]
@pranavrajs done

File: app/models/conversation.rb [hunk 1]
```diff
@@ -25,57 +25,74 @@ class Conversation < ApplicationRecord
...
   end
 
   def toggle_status
-    if open?
-      self.status = :resolved
-    else
-      self.status = :open
-    end
-    self.save! ? true : false
+    self.status = if open?
```
Comment (sojan-official) [inline app/models/conversation.rb]: having if block in assignment  isn't really readable. please use the ternary operator
```suggestion
    self.status = open? ? :resolved : :open
```

File: app/models/conversation.rb [hunk 2]
```diff
@@ -25,57 +25,74 @@ class Conversation < ApplicationRecord
...
+                    :resolved
+                  else
+                    :open
+                  end
+    save! ? true : false
```
Comment (sojan-official) [inline app/models/conversation.rb]: save! would either return true or throw an error.  So there isn't much point in having the true or false statement here 

```suggestion
    save! 
```

File: app/models/conversation.rb [hunk 3]
```diff
@@ -155,24 +162,14 @@ def notify_status_change
     dispatcher_dispatch(ASSIGNEE_CHANGED) if assignee_id_changed?
   end
 
-  def resolve_conversation
```
Comment (sojan-official) [inline app/models/conversation.rb]: why did we remove this method ?

Comment (MatheusRich) [reply app/models/conversation.rb]: Whoop. Just a mistake on the rebase

**Comment by sojan-official** [comment]
@MatheusRich could you also do a rebase, since I can see some conflicts with the latest master

**Comment by MatheusRich** [comment]
@sojan-official I think everything was already address and this doesn't add nothing at all.

**Comment by sojan-official** [comment]
Closing this pull since other PRs fixed this

**Pull request closed without merging**